### PR TITLE
Merge existing registry with default one or configure default metric registry

### PR DESCRIPTION
**Pull request by robert3005**
Lets users plug into spark metrics via default metric registry

### CODE REVIEW

**Comment by robert3005** [comment]
Will submit upstream together with codahale bump

File: core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala [hunk 1]
```diff
@@ -79,6 +79,13 @@ private[spark] class MetricsSystem private (
   private val sources = new mutable.ArrayBuffer[Source]
   private val registry = new MetricRegistry()
 
+  try {
+    registry.registerAll(SharedMetricRegistries.getDefault)
```
Comment (ash211) [inline core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala]: has the default shared metric registry had all the metrics already registered with it at this point in code execution?  it looks like this is a one-time registration so metrics registered with the shared registry afterwards would not be added to Spark's registry

Comment (robert3005) [inline core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala]: @ash211 added listeners to propagate metrics from sources and default registry. This should behave correctly or am I misunderstanding?

**Comment by robert3005** [comment]
Created https://github.com/apache/spark/pull/18406 for upstream

**Pull request merged**